File: crm/backend/src/validateEnv.js
/**
 * Validação obrigatória de variáveis de ambiente no startup.
 * Deve ser chamada antes de iniciar o servidor.
 */
function validateEnv() {
  const env = process.env.NODE_ENV || 'development';
  const jwtSecret = process.env.JWT_SECRET || 'changeme';
  const databaseUrl = process.env.DATABASE_URL || '';
  const useMock = process.env.USE_MOCK !== 'false';

  console.info(`[ENV] Ambiente: ${env}`);
  console.info(`[ENV] USE_MOCK: ${useMock}`);
  console.info(`[ENV] DATABASE_URL definida: ${Boolean(databaseUrl)}`);

  const errors = [];

  if (env === 'production') {
    if (!jwtSecret || jwtSecret === 'changeme') {
      errors.push("JWT_SECRET não pode ser 'changeme' ou vazio em produção.");
    }

    if (!databaseUrl) {
      errors.push('DATABASE_URL é obrigatória em produção.');
    }

    if (useMock) {
      errors.push('USE_MOCK deve ser false em produção.');
    }
  }

  if (errors.length > 0) {
    for (const err of errors) {
      console.error(`[ENV] ERRO: ${err}`);
    }
    console.error('[ENV] Ambiente inválido para produção. Corrija as variáveis acima e reinicie.');
    process.exit(1);
  }

  console.info('[ENV] Validação de ambiente concluída com sucesso.');
}

module.exports = { validateEnv };

File: backend/validate_env.py
"""Validação obrigatória de variáveis de ambiente no startup."""
import os
import sys

from logging_manager import app_logger


def validate_env() -> None:
    env = os.getenv("NODE_ENV", "development")
    jwt_secret = os.getenv("JWT_SECRET", "changeme")
    database_url = os.getenv("DATABASE_URL", "")
    use_mock = os.getenv("USE_MOCK", "false").lower() in ("1", "true", "yes")

    app_logger.info(
        "Validação de ambiente iniciada",
        extra={"env": env, "use_mock": use_mock, "database_url_set": bool(database_url)},
    )

    errors: list[str] = []

    if env == "production":
        if not jwt_secret or jwt_secret == "changeme":
            errors.append("JWT_SECRET não pode ser 'changeme' ou vazio em produção.")

        if not database_url:
            errors.append("DATABASE_URL é obrigatória em produção.")

        if use_mock:
            errors.append("USE_MOCK deve ser false em produção.")

    if errors:
        for err in errors:
            app_logger.error(f"[ENV] {err}")
        app_logger.error(
            "Ambiente inválido para produção. Corrija as variáveis acima e reinicie."
        )
        sys.exit(1)

    app_logger.info(
        "Validação de ambiente concluída com sucesso",
        extra={"env": env},
    )
